File: 2279. Maximum Bags With Full Capacity of Rocks/Solution.py
from typing import List


class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
        differences = []
        for i in range(len(capacity)):
            differences.append(capacity[i]-rocks[i])
        differences.sort()
        sum = 0
        for difference in differences:
            additionalRocks = additionalRocks-difference
            if (additionalRocks >= 0):
                sum = sum+1
            else:
                break
        return sum

File: 2279. Maximum Bags With Full Capacity of Rocks/Solution.java
import java.util.Arrays;

class Solution {
    public int maximumBags(int[] capacity, int[] rocks, int additionalRocks) {
        int diff[] = new int[capacity.length];
        int sum = 0;
        for (int i = 0; i < diff.length; i++) {
            diff[i] = capacity[i] - rocks[i];
        }
        Arrays.sort(diff);
        for (int i = 0; i < diff.length; i++) {
            additionalRocks -= diff[i];
            if (additionalRocks >= 0) {
                sum++;
            } else
                break;
        }
        return sum;
    }
}